### PR TITLE
Rewrite using actix

### PR DESCRIPTION
**Pull request by Geometrically**
Fixes MOD-381

### CODE REVIEW

File: src/stages/player_info.rs [hunk 1]
```diff
@@ -1,37 +1,33 @@
...
 pub struct PlayerInfo {
-    name: String,
+    pub id: String,
+    pub name: String,
 }
 
 impl Default for PlayerInfo {
     fn default() -> Self {
         Self {
+            id: "2578615362f649b0a43b6dfffcf44d33".to_string(),
```
Comment (thesuzerain) [inline src/stages/player_info.rs:16]: Is this really a default player id? Or is it unreachable/a testing string?

File: src/routes/auth.rs [hunk 2]
```diff
@@ -1,148 +1,128 @@
...
+use actix_web::http::StatusCode;
+use serde::Deserialize;
+use serde_json::json;
+use url::Url;
 use uuid::Uuid;
+use crate::db::RuntimeState;
 
 macro_rules! ws_conn_try {
-    ($ctx:literal $status:path, $res:expr => $conn:expr, $ws_conn:expr) => {
+    ($ctx:literal $status:path, $res:expr => $ws_conn:expr) => {
```
Comment (thesuzerain) [inline src/routes/auth.rs:13]: What does this macro do? it seems basically just a function. It it just a stylistic thing?

Comment (Geometrically) [reply src/routes/auth.rs:13]: it's for returning errors for both a websocket and the main file. it's pretty hard to make this a function (don't know if it's possible) due to the varying types which don't share traits involved

File: src/stages/player_info.rs (hunk 1 above)
Comment (Geometrically) [reply src/stages/player_info.rs:16]: unreachable, it's my UUID "Geometrically"

**Review by thesuzerain — APPROVED** [review]

**Pull request merged**